### PR TITLE
License storage related methods extracted into their own class

### PR DESCRIPTION
**Pull request by agr**
Progress on #6533. Addresses https://github.com/NuGet/Engineering/issues/1884.

Extracted the license-related methods into their own storage abstraction class, so it can be used on its own in Orchestrator.

Had to pretty much rewrite on of the `StorageDependentFacts` tests to accommodate the change. Need to confirm with @joelverhagen  that I understood the intention properly.

### CODE REVIEW

File: tests/NuGetGallery.Facts/App_Start/StorageDependentFacts.cs [hunk 1]
```diff
@@ -17,27 +17,45 @@ public void AllTypesDependingOnIFileStorageServiceAreReturnedByGetStorageDepende
...
+                    .Where(gct => gct
+                        .GetConstructors()
+                        .Any(c => c
+                            .GetParameters()
+                            .Any(pi => typeof(ICoreFileStorageService).IsAssignableFrom(pi.ParameterType))))
+                     .Where(gct => !nonStorageTypes.Contains(gct)));
+
+            var fileStorageDependentsInterfaces = new HashSet<Type>(fileStorageDependents.SelectMany(t => t.GetInterfaces()));
+
+            // file storage depdendets that are actually used by Gallery types.
```
Comment (dtivel) [inline tests/NuGetGallery.Facts/App_Start/StorageDependentFacts.cs]: dependents

File: src/NuGetGallery/Configuration/IAppConfiguration.cs [hunk 2]
```diff
@@ -75,6 +75,11 @@ public interface IAppConfiguration : IMessageServiceConfiguration
         /// </summary>
         string AzureStorage_Uploads_ConnectionString { get; set; }
 
+        /// <summary>
+        /// The Azure storage connection string used for RevalidateCertificate job admin panel.
+        /// </summary>
+        string AzureStorage_RevalidateCertificate_ConnectionString { get; set; }
```
Comment (loic-sharma) [inline src/NuGetGallery/Configuration/IAppConfiguration.cs]: >AzureStorage_RevalidateCertificate_ConnectionString [](start = 15, length = 51)

This should be the `AzureStorage_Revalidation_ConnectionString`

**Comment by agr** [comment]
Need to update the deployment configuration to include new configuration key.

**Comment by agr** [comment]
OK, this comes to its final. Need another round of reviews. PR for configuration updates is out.

**Pull request merged**